src/classes/HotkeysUtil.js

```diff
@@ -31,52 +31,164 @@ export default class HotkeysUtil {
       rotateR: 'Rotate Right',
       rotateL: 'Rotate Left',
       resetViewport: 'Reset',
-      clearTools: 'Clear Tools'
-    };
+      clearTools: 'Clear Tools',
+    }
 
     this.commands = {
       scrollDown: {
         name: 'Scroll Down',
-        action: () => log.warn('TODO: scroll down')
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'scrollDown')),
       },
       scrollUp: {
         name: 'Scroll Up',
-        action: () => log.warn('TODO: scroll up')
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'scrollUp')),
       },
       scrollFirstImage: {
         name: 'Scroll to First Image',
-        action: () => log.warn('TODO: scroll to first image')
+        action: () =>
+          this._dispatchCommand(
+            this._getHotKeyCommand(null, 'scrollFirstImage')
+          ),
       },
       scrollLastImage: {
         name: 'Scroll to Last Image',
-        action: () => log.warn('TODO: scroll last image')
+        action: () =>
+          this._dispatchCommand(
+            this._getHotKeyCommand(null, 'scrollLastImage')
+          ),
       },
       previousDisplaySet: {
         name: 'Previous Series',
-        action: () => console.log('TODO: previous series'),
-        disabled: () => !this._canMoveDisplaySets(false)
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'previousSeries')),
+        disabled: () => !this._canMoveDisplaySets(false),
       },
       nextDisplaySet: {
         name: 'Next Series',
-        action: () => console.log('TODO: next display set'),
-        disabled: () => !this._canMoveDisplaySets(true)
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'nextSeries')),
+        disabled: () => !this._canMoveDisplaySets(true),
       },
       nextPanel: {
         name: 'Next Image Viewport',
-        action: () => log.warn('TODO: nextpanel')
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'nextPanel')),
       },
       previousPanel: {
         name: 'Previous Image Viewport',
-        action: () => log.warn('TODO: previous panel')
-      }
-    };
+        action: () =>
+          this._dispatchCommand(this._getHotKeyCommand(null, 'previousPanel')),
+      },
+    }
+  }
+
+  _rotate(currentPosition, rotationDegree) {
+    return currentPosition === undefined
+      ? 0 + rotationDegree
+      : currentPosition + rotationDegree
+  }
+
+  _dispatchCommand(options) {
+    const { setActiveViewportSpecificData } = actions
+    window.store.dispatch(
```

Comment:
Is this the best/only way to access the store?

Reply:
Actually, I am not 100% sure about it either. We can catch up in order to design the best approach.

Reply:
@biharck we should never use `window.store` in this package.

We should get a callback and trigger that callback, so let the caller decide what to do with it. See how [Hanging Protocol Engine](https://github.com/OHIF/ohif-core/blob/63690998e069c5f2027841a08485afd643aebe00/src/hanging-protocols/ProtocolEngine.js#L693) do that.